libexec/qni_symbolic_run.py

```
@@ -451,12 +451,35 @@ def controlled_gate_matrix(qubits: int, controls, target: int, target_gate: Matr
+            first_bit = basis_bit(basis_index, first_target, qubits)
+            second_bit = basis_bit(basis_index, second_target, qubits)
+            output_index = replace_basis_bit(output_index, first_target, qubits, second_bit)
+            output_index = replace_basis_bit(output_index, second_target, qubits, first_bit)
+        matrix[output_index, basis_index] = 1
+
+    return matrix
+
+
 def column_gate_matrix(col, qubits, variables):
     if len(col) != qubits:
         raise ValueError(f"gate column width {len(col)} does not match qubit count {qubits}")
```

Ruff (0.16.3)

[warning] Avoid specifying long messages outside the exception class

(TRY003)